src/test/java/util/xcmailr/XcMailRetrieveEmailsTest.java

```diff
@@ -0,0 +1,126 @@
+package util.xcmailr;
+
+import java.io.IOException;
+
+import javax.mail.MessagingException;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import net.minidev.json.parser.ParseException;
+import util.xcmailr.data.EmailAccount;
+import util.xcmailr.util.SendEmail;
+
+public class XcMailRetrieveEmailsTest extends AbstractXcMailrApiTest
```

Comment:
* [ ] Add Javadoc comments to explain what your test methods want to validate because it is not obvious.
* [ ] It may also make sense to comment the calls to `fetchEmails` and describe what you set.
* [ ] You do not test that `fetchEmail` works with subject and sender. While you do test the shortcut methods, we should also check the `fetchEmail` method works when given these parameters.

Reply:
wouldn't the test for `fetchEmails` with subject and sender parameters be duplicates of the `testRetrieveLastEmailBySubject` and `testRetrieveLastEmailBySender` tests?

Reply:
If we assume white box testing, then yes. Yet, implementations may change to it would be safer to test each functionality. However, you are right, we can assume that the methods will stay the same.